action/protocol/poll/staking_command.go

```diff
@@ -53,16 +51,16 @@ func (sc *stakingCommand) CreateGenesisStates(ctx context.Context, sm protocol.S
 	return sc.stakingV2.CreateGenesisStates(ctx, sm)
 }
 
-func (sc *stakingCommand) Start(ctx context.Context) error {
+func (sc *stakingCommand) Start(ctx context.Context, sr protocol.StateReader) (interface{}, error) {
 	if sc.stakingV1 != nil {
-		if starter, ok := sc.stakingV1.(lifecycle.Starter); ok {
-			return starter.Start(ctx)
+		if starter, ok := sc.stakingV1.(protocol.Starter); ok {
+			if _, err := starter.Start(ctx, sr); err != nil {
+				return nil, err
+			}
 		}
 	}
-	if starter, ok := sc.stakingV2.(lifecycle.Starter); ok {
-		return starter.Start(ctx)
-	}
-	return nil
+	// stakingV2 Start() do nothing now
```

Comment:
i think it's cleaner not to call V2 Start(), since it does nothing now

Reply:
v2 is a protocol, which could be a starter. As a container, sc does not know the details whether it does nothing or not. leave the code as it is.